GUI/dashboard.py

```diff
@@ -1,49 +1,176 @@
 from tkinter import *
 #import <api>
 from PIL import ImageTk, Image
+
+from tkinter import ttk
+from tkinter import filedialog
+from openpyxl import load_workbook
+
+from tkinter.font import Font
+from tkinter.ttk import Style
+
+# setting the path to parent directory so as to import zip
+import sys
+sys.path.insert(0,'..')
+
+from zip import main
+
 root = Tk()
 
 root.title("Check Validity")
 
-canvas1 = Canvas(root, width = 400, height = 300)
-canvas1.pack()
+root.geometry("1250x700")
+
+
+## Adding ScrollBar
+
+main_frame = Frame(root)
+main_frame.pack(fill=BOTH, expand=1)
+
+my_canvas = Canvas(main_frame)
+my_canvas.pack(side=LEFT, fill=BOTH, expand=1)
+
+my_scrollbar = ttk.Scrollbar(main_frame, orient=VERTICAL, command = my_canvas.yview)
+my_scrollbar.pack(side=RIGHT, fill=Y)
+
+my_canvas.configure(yscrollcommand=my_scrollbar.set)
+my_canvas.bind('<Configure>', lambda e: my_canvas.configure(scrollregion = my_canvas.bbox("all")))
+
+# this second_frame is the new 'root' now.
+second_frame = Frame(my_canvas)
+
+my_canvas.create_window((0,0), window=second_frame, anchor="nw")
+
+
+def fileDownload(f):
+	f.save('output.xlsx')
+
+def fileDialog():
+	filename = filedialog.askopenfilename(initialdir = "/", title = "Select an Excel File", filetype = (("xlsx", "*.xlsx"), ("All Files", "*.*")))
+	label_3.config(text=filename)
+
+	try:
+
+		if e.get().isnumeric():
+			main(filename, int(e.get()))
+		else:
+			raise ValueError
+
+		wb = load_workbook(filename)
+		sh = "Sheet"
+		sh = sh + str(e.get())
+		sheet = wb[sh]
+
```

Comment:
`zip.GUI(sh,filename)` 
#this might work

Reply:
Why have you added the funciton GUI ? I was passing the parameters to the main function before.

Reply:
I think you took care of this 🖌️

Reply:
Yes, I told you, I have made changes in valid_ip.py and zip.py to compensate with the dashboard.py code.

But there are some other errors in the zip.py script. They can be figured out by compiling it individually and passing the filename and sheet name in the __main__ .